src/utils/file.ts

```diff
@@ -51,3 +56,82 @@ export async function getEditorOrFileContents(uri: vscode.Uri): Promise<LoadedDo
     throw new Error(`Failed to read file ${uri.toString()}: ${e}`, { cause: e });
   }
 }
+
+export class WriteableTmpDir {
+  static instance: WriteableTmpDir | undefined;
+
+  static getInstance(): WriteableTmpDir {
+    if (!WriteableTmpDir.instance) {
+      WriteableTmpDir.instance = new WriteableTmpDir();
+    }
+    return WriteableTmpDir.instance;
+  }
+
+  /** As determined by {@link determine} */
+  private _tmpdir: string | undefined;
+
+  private constructor() {
+    // Private constructor to prevent external instantiation
+  }
+
+  /**
+   * Determine a writeable temporary directory. This is a best-effort attempt.
+   *
+   * Should be called at extension startup to make subsequent calls to
+   * {@link get}.
+   *
+   * (We have reports that when installed through JamfAppInstallers on OSX, tmpdir() is not actually writeable.)
+   */
+  async determine(): Promise<void> {
+    const possibleDirs = [
+      tmpdir(), // Should work on all platforms, but JamfAppInstallers on OSX may mangle?
+      process.env["TMPDIR"], // UNIX-y, but should also have been what tmpdir() returned.
+      process.env["TEMP"], // Windows-y, probably also what tmpdir() returns on Windows.
+      process.env["TMP"], // sometimes Windows-y
+      "/var/tmp", // UNIX-y
+      "/tmp", // UNIX-y
+      "/private/tmp", // macOS
+    ];
+    const errorsEncountered: Error[] = [];
+
+    for (const dir of possibleDirs) {
+      if (!dir) {
+        continue; // Skip undefined or null directories
+      }
+      try {
+        // Check if the directory is writeable
+        const fileUri = vscode.Uri.file(join(dir, ".vscode_test.tmp"));
+        await writeFile(fileUri, Buffer.from("test"));
+        await deleteFile(fileUri);
+        this._tmpdir = dir;
+        logger.info(`Found writeable tmpdir: ${dir}`);
+        return;
+      } catch (e) {
+        logger.warn(`Failed to write to ${dir}: ${e}`);
+        errorsEncountered.push(e as Error);
+        // Ignore errors and try the next directory
+      }
+    }
+
+    logError(
+      logger,
+      "determineWriteableTmpDir(): No writeable tmpdir found.",
```

Comment:
First argument here should be an `Error` (or subclass), so I think we'd need this for sending to Sentry:
```suggestion
      new Error("determineWriteableTmpDir(): No writeable tmpdir found."),
      "sidecar log",
```

Reply:
thanks! Fix coming up.